src/components/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { useEffect, useRef, useState } from 'react';
+import classnames from 'classnames';
+
+import './index.scss';
+
+type TabsProps = {
+  defaultActiveTab?: string;
+  children: React.ReactElement[];
+};
+
+const DIRECTION: { [key: string]: number } = {
+  ArrowLeft: -1,
+  ArrowUp: -1,
+  ArrowRight: 1,
+  ArrowDown: 1
+};
+
+const KEYS = {
+  ARROW_LEFT: 'ArrowLeft',
+  ARROW_UP: 'ArrowUp',
+  ARROW_RIGHT: 'ArrowRight',
+  ARROW_DOWN: 'ArrowDown',
+  HOME: 'Home',
+  END: 'End'
+};
+
+// TODO: Responsive Design
+// Need to stack tabs and tab panels
+// Disable left/right arrows and use up down
+// Need to use aria-direction
+
+const Tabs = ({ defaultActiveTab, children }: TabsProps) => {
+  const tabsRef = useRef<HTMLUListElement>(null);
+  const [tabEls = [], setTabEls] = useState<NodeList>();
+  const tabObjs = children.map(child => ({
+    id: child.props.id,
+    name: child.props.tabName
+  }));
+  const [activeTab, setActiveTab] = useState(
+    defaultActiveTab || tabObjs[0].name
+  );
+
+  const handleKeyup = (e: KeyboardEvent) => {
```

Comment:
Why are we using both keyUp and keyDown? Can we do them all with keyUp?

Reply:
That's how it's done on the W3C example.

The left/right arrows are on key up to prevent the tabs from constantly switching in the case left/right arrows are held down.

The home/end buttons are on key down so the switch tabs when keys are clicked and don't need to wait until the user releases the key. If the keys are  held down (accidentally), the tabs don't rapidly switch.